### ISSUE TITLE
Cannot access Teleport via Github

### ISSUE THREAD

**Issue by Chuyentest123** [gravitational/teleport#27097]
Hope you answer my issue soon.
I just do as the document but I can't access Teleport via GitHub
Here is my config file at Teleport and GitHub.yaml
![1](https://github.com/gravitational/teleport/assets/128662730/827b941d-e543-4b38-8edb-6a93dcc1a37d)

![2](https://github.com/gravitational/teleport/assets/128662730/1de62c15-84ac-4d2d-8e47-cc2d9cc66c64)

And here is my organization and my Team:
![image](https://github.com/gravitational/teleport/assets/128662730/c64ce3c3-0b10-479f-baf2-0497fb4517d1)
![image](https://github.com/gravitational/teleport/assets/128662730/970c02d5-9359-4a58-88f8-9353a4a1338a)

And the OAuthApps:
![image](https://github.com/gravitational/teleport/assets/128662730/e0ed9f1e-d3fa-4df1-9e7a-4e1fbcc2f898)

Here is the progress of login:
1. This is Chuyentest123's GitHub account  (Administrator of the organization and member of TestEditor):
![image](https://github.com/gravitational/teleport/assets/128662730/fb17b5dc-852f-47cd-b770-9fa3ff359dc6)

2. After login GitHub success I have the notification like this (I could not take the photo of Chuyentest123 so I took this photo instead) :
![image](https://github.com/gravitational/teleport/assets/128662730/55da06b6-6abd-4a9c-bc09-d6840cc9abf6)

3. And after clicking authorize my-user, it failed:
![image](https://github.com/gravitational/teleport/assets/128662730/87c77af3-4310-48dd-8e5d-9e929bbac1f0)

Here is the log of the failed login:
![image](https://github.com/gravitational/teleport/assets/128662730/dab45de0-59fd-4be2-8601-9aaec1dc44a8)

Expected behavior:
I can access Teleport via GitHub like this
![image](https://github.com/gravitational/teleport/assets/128662730/2a0848fe-848b-4611-b17b-90cd7917f9c1)

Current behavior:
![image](https://github.com/gravitational/teleport/assets/128662730/58d81c51-317f-4a07-85c3-708e26c84774)

Bug details:
- Teleport version: 11.4.4
- Recreation steps
- Debug logs

**Comment by stevenGravy** [comment]
@Chuyentest123 the team is `testauditor` which doesn't have a match in the connector.  The case is different too.

**Comment by Chuyentest123** [comment]
> @Chuyentest123 the team is `testauditor` which doesn't have a match in the connector. The case is different too.

@stevenGravy  No. Chuyentest123 is in "testEditor" team. The testAuditor is different.
